Sources/MapboxMaps/Foundation/Camera/CameraManager.swift

```diff
@@ -140,31 +132,38 @@ public class CameraManager {
             return
         }
 
+        internalAnimator?.stopAnimation()
+
         let clampedCamera = CameraOptions(center: targetCamera.center,
                                           padding: targetCamera.padding,
                                           anchor: targetCamera.anchor,
                                           zoom: targetCamera.zoom?.clamped(to: mapCameraOptions.minimumZoomLevel...mapCameraOptions.maximumZoomLevel),
-                                          bearing: optimizeBearing(startBearing: mapView.cameraView.localBearing, endBearing: targetCamera.bearing),
+                                          bearing: targetCamera.bearing,
                                           pitch: targetCamera.pitch?.clamped(to: mapCameraOptions.minimumPitch...mapCameraOptions.maximumPitch))
 
         // Return early if the cameraView's camera is already at `clampedCamera`
         guard mapView.cameraOptions != clampedCamera else {
             return
         }
 
-        let transitionBlock = {
-            mapView.cameraOptions = clampedCamera
-        }
-
         if animated && duration > 0 {
-            performCameraAnimation(duration: duration, animation: transitionBlock, completion: completion)
+            let animation = { (transition: inout CameraTransition) in
+                transition.center.toValue = clampedCamera.center
+                transition.padding.toValue = clampedCamera.padding
+                transition.anchor.toValue = clampedCamera.anchor
+                transition.zoom.toValue = clampedCamera.zoom
+                transition.bearing.toValue = clampedCamera.bearing
+                transition.pitch.toValue = clampedCamera.pitch
+            }
+            performCameraAnimation(duration: duration, animation: animation, completion: completion)
         } else {
-            transitionBlock()
+            mapView.mapboxMap.updateCamera(with: clampedCamera)
         }
     }
 
     public func cancelAnimations() {
-        for animator in cameraAnimators.allObjects where animator.state == .active {
+        guard let validMapView = mapView else { return }
+        for animator in validMapView.cameraAnimatorsHashTable.allObjects where animator.state == UIViewAnimatingState.active {
```

Comment:
could developers restart the animation? what does that mean for gestures?

Reply:
Canceled animations are not meant to be restarted, in the current design. Will add docs to reflect that.

Reply:
does anything bad happen if we remove the `where` clause?

Reply:
from discussion: remove where clause